src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/semantic/node/model/ModelImplWrapper.java

```diff
@@ -26,11 +25,13 @@ public ModelImplWrapper(String name, ModelImpl model) {
 
     @SuppressWarnings("unchecked")
     public List<String> getRequired() {
+        List<String> requiredProperties;
         try {
-            return emptyIfNull((List<String>) readField(model, "required", true));
+            requiredProperties = new ArrayList<>(emptyIfNull((List<String>) readField(model, "required", true)));
         } catch (IllegalAccessException ex) {
-            return emptyList();
+            requiredProperties = new ArrayList<>();
         }
+        return requiredProperties;
```

Comment:
What is purpose of the refactoring? I'm assuming that you was need to update returned list even in case if it was returned without entries, am I right?

Reply:
Your assumption is right